kong/db/dao/wasm_filter_chains.lua

```diff
@@ -0,0 +1,108 @@
+local wasm_filter_chains = {}
+
+local insert = table.insert
+local fmt = string.format
+
+local EMPTY = {}
+
+
+local function check_enabled_filters(self, chain)
+  if not self.filters then
+    local err_t = self.errors:schema_violation({
+      filters = "no wasm filters are configured",
+    })
+    return nil, tostring(err_t), err_t
+  end
+
+  if type(chain.filters) ~= "table" then
+    return true
+  end
+
+  local errs
+
+  for i, filter in ipairs(chain.filters) do
+    local name = filter.name
+
+    -- let the standard schema validation catch invalid name errors
+    if type(name) == "string"
+       and not self.filters_by_name[name]
+    then
+      errs = errs or {}
+      errs[i] = { name = "no such filter: " .. filter.name }
+    end
+  end
+
+  if errs then
+    local err_t = self.errors:schema_violation({
+      filters = errs,
+    })
+    return nil, tostring(err_t), err_t
+  end
+
+  return true
+end
+
+
+function wasm_filter_chains:load_filters(wasm_filters)
+  local filters = {}
+  local filters_by_name = {}
+
+  local errors = {}
+
+  for i, filter in ipairs(wasm_filters or EMPTY) do
+    insert(filters, filter)
+
+    if type(filter.name) ~= "string" then
+      insert(errors, fmt("filter #%s name is not a string", i))
+
+    elseif filters_by_name[filter.name] then
+      insert(errors, fmt("duplicate filter name (%s) at #%s", filter.name, i))
+
+    else
+      filters_by_name[filter.name] = filter
+
+    end
+  end
+
+  if #errors > 0 then
+    return nil, "failed to load filters: " .. table.concat(errors, ", ")
```

Comment:
use 

```lua
local concat = table.concat
```

at the module header is better

Reply:
Thanks! I've left most optimizations out of this PR for simplicity's sake. I will take care of these things in a follow-up PR.